pkg/migrate/migrate.go

```diff
@@ -41,13 +48,30 @@ var isDestScLocalVolumeProvisioner bool
 
 // Options is the set of options that should be provided to Migrate
 type Options struct {
-	SourceSCName         string
-	DestSCName           string
-	RsyncImage           string
-	Namespace            string
-	SetDefaults          bool
-	VerboseCopy          bool
-	SkipSourceValidation bool
+	SourceSCName               string
+	DestSCName                 string
+	RsyncImage                 string
+	Namespace                  string
+	SetDefaults                bool
+	VerboseCopy                bool
+	SkipSourceValidation       bool
+	SkipPVAccessModeValidation bool
+	PvcCopyTimeout             int
+	PodReadyTimeout            int
+}
+
+// PVMigrator represents a migration context for migrating data from all srcSC volumes to
+// dstSc volumes.
+type PVMigrator struct {
```

Comment:
When adding support for this, I needed some contextual data during the migration behavior and so the struct was born. I can make  `Migrate()` into a method instead of a function.

Reply:
This struct is at the very least poorly named
it is not actually related to migrating PVs
it's only used for determining whether volume access modes work

IMO we should just be passing the relevant things to the function, because the only things that aren't already included in 'options' and name overrides